Add release notes for 0.12.0 release (#732)
submariner-io/submariner-website: src/content/community/releases/_index.en.md

weight = 40
+++

## v0.12.0

**New features**

* Added a new `subctl uninstall` command that removes all Submariner components and dataplane artifacts, such as IP table rules, routes from
a cluster.
* Added a new `subctl unexport` command that stops exporting a previously exported service.
* Added new `subctl cloud prepare` and `subctl cloud cleanup` commands for the Red Hat OpenStack Platform (RHOS).
* New Globalnet metrics were added:

>   * Count of global Egress IPs allocated at Cluster scope, namespace scope and for selected pods per CIDR.

Reviewer: ```suggestion
  * Count of global Egress IPs allocated at Cluster scope, namespace scope, and for selected pods per CIDR.
```

I think a serial comma would reduce ambiguity here, to make it clear that "namespace scope and for selected pods per CIDR" are not more closely related than "Cluster scope, namespace scope". I think this is a case of point 5 here: https://en.wikipedia.org/wiki/Serial_comma#Arguments_for_and_against

  * Count of global Ingress IPs allocated for Pods/Services per CIDR.
* Added support for Globalnet objects verification using subctl diagnose command.
* Added support for `--broker-namespace` flag while deploying the Broker.
* Added support for running `subctl diagnose` on single node clusters.
* Added support for running `subctl diagnose` from a pod in cluster.

> * `subctl cloud prepare` now deploys a dedicated gateway node as a default option on GCP and OpenStack platforms.

Reviewer: I'm wondering if the K8s docs guidance to avoid "starting a sentence with a component tool or component name" applies here.

https://kubernetes.io/docs/contribute/style/style-guide/#starting-a-sentence-with-a-component-tool-or-component-name

I'm okay with keeping it as-is because it's easy to parse visually.

* `subctl show` command now shows information about the Broker CR in the cluster.

> * `subctl gather` now collects the necessary globalnet objects like clusterGlobalEgressIPs, globalEgressIPs, globalIngressIPs.

Reviewer 2: ```suggestion
* `subctl gather` now collects Globalnet information.
```

> * `subctl gather` now collects the necessary globalnet objects like clusterGlobalEgressIPs, globalEgressIPs, globalIngressIPs.

Reviewer: ```suggestion
* `subctl gather` now collects the necessary Globalnet objects like clusterGlobalEgressIPs, globalEgressIPs, globalIngressIPs.
```

If GitHub's infra wasn't broken and keeping our CI from running, I think linting would flag this:

https://github.com/submariner-io/submariner-website/blob/devel/.markdownlint.yml#L25

> * `subctl diagnose` throws a warning when a generic CNI network plugin is detected.

Reviewer 2: ```suggestion
* `subctl diagnose` displays a warning when a generic CNI network plugin is detected.
```

* New Service Discovery metric was added:
  * Count of DNS queries handled by Lighthouse plugin.

**Bug fixes**

* Calico is now correctly detected when used as a third-party plugin in OpenShift.
* Services without selectors can now be resolved across the ClusterSet.
* `subctl diagnose firewall inter-cluster` command will now work correctly for vxlan cable driver.

> **Internal changes**

Reviewer 2: The RN notes should reflect user-facing changes so I don't think we should use the term "Internal". We could name it "Other changes" but I think the 2 items below can qualify as "New features" (perhaps depending on POV).  I mean I think anything not a bug can be considered as a new feature. Eg, if L25 above is considered a new feature then why wouldn't also the  2 items below?

The other option is to not categorize the items as we didn't for past releases. (Do we want to specifically call out and highlight bug fixes 😄)  Perhaps we keep the "Bug fixes" category but list everything else uncategorized.

WDYT @skitt @sridhargaddam @dfarrell07 @vthapar @nyechiel et al

* The broker token and IPsec PSK are now stored in secrets which are used in preference to the corresponding fields in the Submariner CR,
which are now deprecated. For backwards compatibility and to simplify upgrades, the deprecated fields are still populated but will be
removed in 0.13.
* Globalnet no longer uses `kube-proxy` chains in support of exported services. Instead, it now creates an
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service since some deployments of
Kubernetes does not allow Services with `ExternalIPs` to be created for security reasons. The prerequisites outlined in

> the [Globalnet page](https://submariner.io/getting-started/architecture/globalnet/) need to be configured.

Reviewer: ```suggestion
* Globalnet no longer uses `kube-proxy` chains in support of exported services. Instead, it now creates an
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service. Some Kubernetes distributions don't allow Services with `ExternalIPs` by default for security reasons. Users must follow the (Globalnet prerequisites)[link] to allow the Globalnet controller to create/update/delete Services with ExternalIPs.
```

I'm assuming `ExternalIPs` is a K8s object field? That's why I'm using `code case`.

> the [Globalnet page](https://submariner.io/getting-started/architecture/globalnet/) need to be configured.

Reviewer 2: ```suggestion
* Globalnet no longer uses `kube-proxy` chains in support of exported services. Instead, it now creates an
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service. Since some deployments of
Kubernetes do not allow Services with `ExternalIPs` to be created for security reasons. the pre-requisites outlined in
the [Globalnet page](https://submariner.io/getting-started/architecture/globalnet/) need to be configured.
```

**Known Issues:**

> * [When using the dot character in the cluster name, service discovery doesn’t work](https://github.com/submariner-io/lighthouse/issues/707).

Reviewer 2: The long hyperlink looks awkward - how about linking it at the end as we've done elsewhere

```suggestion
* When using the dot character in the cluster name, service discovery doesn’t work ([#707](https://github.com/submariner-io/lighthouse/issues/707)).
```

* On OpenShift, Globalnet metrics do not appear automatically. This can be fixed by manually opening the Globalnet metrics port, TCP/8081.

> * [When using `subctl cloud prepare` on Red Hat OpenStack Platform (RHOS), if dedicated gateway is used, security groups are associated with
> wrong node](https://github.com/submariner-io/cloud-prepare/issues/227).

Reviewer 2: Same here

```suggestion
* When using `subctl cloud prepare` on Red Hat OpenStack Platform (RHOS), if a dedicated gateway is used, the security groups are associated with
the wrong node ([#227](https://github.com/submariner-io/cloud-prepare/issues/227)).
```

weight = 40
+++

## v0.12.0

### New features

> * Added a new `subctl uninstall` command that removes all Submariner components and dataplane artifacts, such as IP table rules, routes from

Reviewer: Maybe something like this would be more clear:

```suggestion
* Added a new `subctl uninstall` command that removes all Submariner components and dataplane artifacts, such as iptables rules and routing table entries, from
```

Regarding "IP table" vs "iptables":

https://submariner.io/development/website/style-guide/#submarinerio-word-list

Reviewer 3: Yes, we should use iptables.

a cluster.
* Added a new `subctl unexport` command that stops exporting a previously exported service.
* Added new `subctl cloud prepare` and `subctl cloud cleanup` commands for the Red Hat OpenStack Platform (RHOS).
* New Globalnet metrics were added:
  * Count of global Egress IPs allocated at Cluster scope, namespace scope, and for selected pods per CIDR.
  * Count of global Ingress IPs allocated for Pods/Services per CIDR.
* Added support for Globalnet objects verification using subctl diagnose command.
* Added support for `--broker-namespace` flag while deploying the Broker.
* Added support for running `subctl diagnose` on single node clusters.
* Added support for running `subctl diagnose` from a pod in cluster.
* `subctl cloud prepare` now deploys a dedicated gateway node as a default option on GCP and OpenStack platforms.

> * `subctl show` command now shows information about the Broker CR in the cluster.

Reviewer 4: ```suggestion
* `subctl show` now shows information about the Broker CR in the cluster.
```

for consistency with the other `subctl …` entries.

* `subctl gather` now collects Globalnet information.
* `subctl diagnose` displays a warning when a generic CNI network plugin is detected.
* New Service Discovery metric was added:

>   * Count of DNS queries handled by Lighthouse plugin.

Reviewer: Can we move this to be by the parallel "New Globalnet metrics were added:" sublist? Otherwise, it doesn't make much sense to break this out as a sublist when it only has one item.

Author: Made it a sentence instead of a list.

Reviewer: Doesn't it seem better to group the "metrics added" points? Why keep them so far apart in the list?

### Bug fixes

> * Calico is now correctly detected when used as a third-party plugin in OpenShift.

Reviewer: ```suggestion
* Calico is now correctly detected when used as a network plugin in OpenShift.
```

* Services without selectors can now be resolved across the ClusterSet.

> * `subctl diagnose firewall inter-cluster` command will now work correctly for VxLAN cable driver.

Reviewer 4: ```suggestion
* `subctl diagnose firewall inter-cluster` command will now work correctly for VXLAN cable driver.
```

> * `subctl diagnose firewall inter-cluster` command will now work correctly for VXLAN cable driver.

Reviewer: This is a pretty clear example of https://kubernetes.io/docs/contribute/style/style-guide/#starting-a-sentence-with-a-component-tool-or-component-name. I think we can do the fix recommended there:

```suggestion
* The `subctl diagnose firewall inter-cluster` command will now work correctly for VXLAN cable driver.
```

Or make the structure parallel to the similar lines above:

```suggestion
* `subctl diagnose firewall inter-cluster` now works correctly for the VXLAN cable driver.
```

### Other changes

> * The broker token and IPsec PSK are now stored in secrets which are used in preference to the corresponding fields in the Submariner CR

Reviewer: ```suggestion
* The broker token and IPsec PSK are now stored in secrets which are used in preference to the corresponding fields in the Submariner CR,
```

Maybe?

which are now deprecated. For backwards compatibility and to simplify upgrades, the deprecated fields are still populated but will be
removed in 0.13.
* Globalnet no longer uses `kube-proxy` chains in support of exported services. Instead, it now creates an

> internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service. Since some deployments of

Reviewer: ```suggestion
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service. Some deployments of
```

This sentence seems like it's missing a part, has a hanging "since". Are we trying to say something like...

"Some Kubernetes distributions don't allow Services with `ExternalIPs` by default for security reasons. Users must follow the (Globalnet prerequisites)[link] to allow the Globalnet controller to create/update/delete Services with `ExternalIPs`."

Author: What is meant is that "Instead, it now creates an
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service since some deployments of
Kubernetes does not allow Services with `ExternalIPs` to be created for security reasons."

Reviewer: I think that only partly address the feedback, see the updated suggestion I sent below that might be more clear.

> Kubernetes does not allow Services with `ExternalIPs` to be created for security reasons. The pre-requisites outlined in

Reviewer: ```suggestion
Kubernetes does not allow Services with `ExternalIPs` to be created for security reasons. The prerequisites outlined in
```

weight = 40
+++

## v0.12.0

**New features**

> * Added a new `subctl uninstall` command that removes all Submariner components and dataplane artifacts, such as IP table rules, routes from

Reviewer 5: Instead of starting with "subctl uninstall", can we move it to Line 30 ;-)

Author: I think its an important and much needed feature addition and should be highlighted ;-)

Reviewer 5: Haha...I'm not against highlighting :D 

I felt that its like showing how to uninstall your product even before talking about the product features. It would be good if we list it at the end (unless there are no other important things to highlight) similar to the way such things are captured in documentation ;-)

Reviewer 5: Anyways, this is more of a nit and I'm fine if others are okay with the current flow.

Reviewer 6: I think this is fine. Those who are already using submariner will find this very useful and this is one of the bigger QoL feature for users, so okay to mention it upfront.

a cluster.
* Added a new `subctl unexport` command that stops exporting a previously exported service.

> * Added new `subctl cloud prepare` and `subctl cloud cleanup` commands for the Red Hat OpenStack Platform (RHOS). This ensures the
> prerequisites for installing Submariner on RHOS are satisfied.

Reviewer 5: ```suggestion
* Added new `subctl cloud prepare` and `subctl cloud cleanup` commands for the Red Hat OpenStack Platform (RHOS). 
```

Reviewer 5: We have a known issue for this, so cannot use the description that "prerequisites for installing Submariner on RHOS are satisfied". We expect the user to manually configure the SG rules.

* New Globalnet metrics were added:

>   * Count of global Egress IPs allocated for Pods/Services per CIDR.
>   * Count of global Egress IPs allocated for clusters per CIDR.

Reviewer 5: ```suggestion
  * Count of global Egress IPs allocated at Cluster scope, namespace scope and for selected pods per CIDR.
```

* Added support for `--broker-namespace` flag while deploying the Broker.
* Added support for running `subctl diagnose` on single node clusters.
* Added support for running `subctl diagnose` from a pod in cluster.
* `subctl cloud prepare` now deploys a dedicated gateway node as a default option on GCP and OpenStack platforms.
* `subctl show` command now shows information about the Broker CR in the cluster.
* `subctl gather` now collects Globalnet information.
* `subctl diagnose` displays a warning when a generic CNI network plugin is detected.
* New Service Discovery metric was added:
  * Count of DNS queries handled by Lighthouse plugin.

### Bug fixes

* Calico is now correctly detected when used as a third-party plugin in OpenShift.
* Services without selectors can now be resolved across the ClusterSet.

> * `subctl diagnose firewall inter-cluster` command will now work correctly for vxlan cable driver.

Reviewer: ```suggestion
* `subctl diagnose firewall inter-cluster` command will now work correctly for VXLAN cable driver.
```

> ### Internal changes

Reviewer 6: ```suggestion
### Other changes
```

Reviewer: Verified this is done :heavy_check_mark:

* The broker token and IPsec PSK are now stored in secrets which are used in preference to the corresponding fields in the Submariner CR
which are now deprecated. For backwards compatibility and to simplify upgrades, the deprecated fields are still populated but will be
removed in 0.13.
* Globalnet no longer uses `kube-proxy` chains in support of exported services. Instead, it now creates an 
internal `ClusterIP` Service with the `ExternalIPs` set to the global IP assigned to the corresponding Service. Since some deployments of 

> Kubernetes do not allow Services with `ExternalIPs` to be created for security reasons. the pre-requisites outlined in

Reviewer: ```suggestion
Kubernetes do not allow Services with `ExternalIPs` to be created for security reasons. The pre-requisites outlined in
```

> Kubernetes do not allow Services with `ExternalIPs` to be created for security reasons. the pre-requisites outlined in

Reviewer 6: ```suggestion
Kubernetes does not allow Services with `ExternalIPs` to be created for security reasons. the pre-requisites outlined in
```

Reviewer: This sentence is now quite different, but the grammar is corrected. :heavy_check_mark:

Kubernetes do not allow the Service with externalIPs to be created for security reasons. To mitigate this, the pre-requisites mentioned in

> the Globalnet page need to be configured.

Reviewer 5: ```suggestion
the [Globalnet page](https://submariner.io/getting-started/architecture/globalnet/) need to be configured.
```

### Known Issues

* When using the dot character in the cluster name, service discovery doesn’t work
([#707](https://github.com/submariner-io/lighthouse/issues/707)).
* On OpenShift, Globalnet metrics do not appear automatically. This can be fixed by manually opening the Globalnet metrics port, TCP/8081.

> * When using `subctl cloud prepare` on Red Hat OpenStack Platform (RHOS), if a dedicated gateway is used, the security groups are

Reviewer: Does this need to be added to the main known issues, or is it already fixed on devel? https://submariner.io/operations/known-issues/

Reviewer 5: The known-issues are still applicable

Author: Should I move all the known issues to the `known issues` section?

Reviewer: I think any that still apply to devel should also be in the main Known Issues.

associated with the wrong node ([#227](https://github.com/submariner-io/cloud-prepare/issues/227)).

> * With Red Hat OpenStack Platform (RHOS), when using a dedicated gateway node the submariner gateway security group and Submariner internal

Reviewer 5: This seems to be related to the above issue.

Reviewer 7: yes they are referring to same issue. Can be combined into one.

> * With Red Hat OpenStack Platform (RHOS), when using a dedicated gateway node the submariner gateway security group and Submariner internal

Reviewer: ```suggestion
* With Red Hat OpenStack Platform (RHOS), when using a dedicated gateway node the Submariner gateway security group and Submariner internal
```

> * With Red Hat OpenStack Platform (RHOS), when using a dedicated gateway node the submariner gateway security group and submariner internal

Reviewer 5: @vthapar is the above point and this one same? If so, can we merge them with appropriate description.
